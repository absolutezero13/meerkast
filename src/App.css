* {
  box-sizing: border-box;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue";
}

a {
  text-decoration: none;
  color: rgb(173, 170, 170);
}
a:hover {
  text-decoration: underline;
}
