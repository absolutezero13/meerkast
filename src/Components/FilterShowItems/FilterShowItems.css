.filter-show-items {
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: space-between;
  padding: 6px 10%;
  padding-bottom: 0;
}
.filter-show-items input,
.filter-show-items select {
  width: 300px;
  height: 30px;
  box-sizing: border-box;
}
.filter-show-items input:focus,
.filter-show-items select:focus {
  box-shadow: 0 0 5px rgba(81, 203, 238, 1);
  padding: 3px 0px 3px 3px;
  border: 1px solid rgba(81, 203, 238, 1);
}
@media only screen and (max-width: 600px) {
  .filter-show-items {
    justify-content: flex-start;
  }
}
@media only screen and (max-width: 360px) {
  .filter-show-items input,
  .filter-show-items select {
    width: 90vw;
  }
  .filter-show-items {
    flex-direction: column;
    align-items: center;
  }
}
