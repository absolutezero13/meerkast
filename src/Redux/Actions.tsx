export const setTextFilter = (text: string = "") => ({
  type: "SET_TEXT_FILTER",
  text,
});
export const setSortByAscendingYear = () => ({
  type: "SORT_BY_ASCENDING_YEAR",
});
export const setSortByDescendingYear = () => ({
  type: "SORT_BY_DESCENDING_YEAR",
});
export const setSortByAscendingTitle = () => ({
  type: "SORT_BY_ASCENDING_TITLE",
});
export const setSortByDescendingTitle = () => ({
  type: "SORT_BY_DESCENDING_TITLE",
});
