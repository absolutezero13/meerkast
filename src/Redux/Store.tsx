import { createStore, combineReducers } from "redux";
import { moviesReducer, filterReducer, seriesReducer } from "./Reducers";

export const store = createStore(
  combineReducers({
    movies: moviesReducer,
    series: seriesReducer,
    filters: filterReducer,
  })
);
