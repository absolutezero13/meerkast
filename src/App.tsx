import React from "react";
import "./App.css";
import HomePage from "./Components/HomePage";
import "./firebase/firebase";

const App: React.FC = () => {
  return (
    <div className="App">
      <HomePage />
    </div>
  );
};

export default App;
