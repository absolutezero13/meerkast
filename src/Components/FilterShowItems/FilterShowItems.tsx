import React from "react";
import { connect } from "react-redux";
import {
  setTextFilter,
  setSortByAscendingTitle,
  setSortByAscendingYear,
  setSortByDescendingTitle,
  setSortByDescendingYear,
} from "../../Redux/Actions";
import "./FilterShowItems.css";

const FilterShowItems: React.FC = (props: any) => {
  return (
    <div className="filter-show-items">
      <input
        placeholder="Search Shows.."
        type="search"
        value={props.filters.text}
        onChange={(e) => props.dispatch(setTextFilter(e.target.value))}
      />
      <select
        onChange={(e) => {
          if (e.target.value === "ascending-year") {
            props.dispatch(setSortByAscendingYear());
          } else if (e.target.value === "descending-year") {
            props.dispatch(setSortByDescendingYear());
          } else if (e.target.value === "ascending-title") {
            props.dispatch(setSortByAscendingTitle());
          } else if (e.target.value === "descending-title") {
            props.dispatch(setSortByDescendingTitle());
          }
        }}
      >
        <option defaultValue="nothing" value="nothing">
          Sort By
        </option>
        <option value="ascending-year">Year (ascending)</option>
        <option value="descending-year">Year (descending)</option>
        <option value="ascending-title">Title (ascending)</option>
        <option value="descending-title">Title (descending)</option>
      </select>
    </div>
  );
};

const mapStateToProps = (state: any) => {
  return {
    filters: state.filters,
  };
};

export default connect(mapStateToProps)(FilterShowItems);
