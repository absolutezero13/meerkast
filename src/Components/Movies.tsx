import { render } from "@testing-library/react";
import React from "react";
import "./MoviesAndSeries.css";
import ListShowItem from "./ListShowItem";
import { getVisibleShows } from "../Redux/Selector";
import { connect } from "react-redux";

const Movies: React.FC = (props: any) => {
  return (
    <div className="movies">
      {props.movies.map((movie: any) => {
        return <ListShowItem key={Math.random()} {...movie} />;
      })}
    </div>
  );
};

const mapStateToProps = (state: any) => {
  return {
    movies: getVisibleShows(state.movies, state.filters),
    filters: state.filters,
  };
};

export default connect(mapStateToProps)(Movies);
