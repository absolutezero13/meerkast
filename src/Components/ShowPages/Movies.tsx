import React from "react";
import "./MoviesAndSeries.css";
import ListShowItem from "../ListShowItem/ListShowItem";
import { getVisibleShows } from "../../Redux/Selector";
import { connect } from "react-redux";
import FilterShowItems from "../FilterShowItems/FilterShowItems";
import { State } from "../../Redux/Reducers";
import { Show } from "../../Redux/Reducers";

const Movies: React.FC<State> = (props) => {
  return (
    <div className="shows">
      <FilterShowItems />
      <div className="show__items">
        {props.movies.map((movie: Show) => {
          return <ListShowItem key={movie.title} {...movie} />;
        })}
      </div>
    </div>
  );
};

const mapStateToProps = (state: State) => {
  return {
    movies: getVisibleShows(state.movies, state.filters),
  };
};

export default connect(mapStateToProps)(Movies);
