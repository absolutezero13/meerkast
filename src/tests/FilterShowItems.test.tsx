import React from "react";
import { render, fireEvent } from "@testing-library/react";
import FilterShowItems from "../Components/FilterShowItems/FilterShowItems";
import App from "../App";
import { testShowData } from "../fixtures/testShowData";
import Movies from "../Components/ShowPages/Movies";
import { screen } from "@testing-library/dom";
import { store } from "./../Redux/Store";
import { Provider } from "react-redux";
// const input: any = screen.getByTestId("search");

test("filter show items match snapshot", () => {
  const { container } = render(
    <Provider store={store}>
      <FilterShowItems placeholder={"Search anything"} />
    </Provider>
  );
  expect(container).toMatchSnapshot();
});

// INPUT ELEMENT TESTING

const setup = () => {
  //no need for destructring here
  const container = render(
    <Provider store={store}>
      <FilterShowItems placeholder={"Search anything"} />
    </Provider>
  );

  const input = container.getByTestId("search") as HTMLInputElement;
  return {
    input,
    ...container,
  };
};
test("it should change input value to hey", () => {
  const { input } = setup();
  fireEvent.change(input, { target: { value: "hey" } });
  expect(input.value).toBe("hey");
});

// SELECT ELEMENT TESTING
const setupSelect = () => {
  const container = render(
    <Provider store={store}>
      <FilterShowItems placeholder={"Search anything"} />
    </Provider>
  );

  const select = container.getByTestId("select") as HTMLSelectElement;
  return {
    select,
    ...container,
  };
};
test("it should change select value to hey", () => {
  const { select } = setupSelect();
  fireEvent.change(select, { target: { value: "ascending-year" } });
  expect(select.value).toBe("ascending-year");
});
