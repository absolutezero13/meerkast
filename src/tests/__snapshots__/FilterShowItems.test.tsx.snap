// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter show items match snapshot 1`] = `
<div>
  <div
    class="filter-show-items"
  >
    <input
      data-testid="search"
      placeholder="Search Search anything 🔍"
      type="search"
      value=""
    />
    <select
      data-testid="select"
    >
      <option
        value="nothing"
      >
        Sort By
      </option>
      <option
        value="ascending-year"
      >
        Year (ascending)
      </option>
      <option
        value="descending-year"
      >
        Year (descending)
      </option>
      <option
        value="ascending-title"
      >
        Title (ascending)
      </option>
      <option
        value="descending-title"
      >
        Title (descending)
      </option>
    </select>
  </div>
</div>
`;
